webapp/src/test/java/com/box/l10n/mojito/monitoring/QuartzPendingJobsReportingTaskTest.java

```diff
@@ -0,0 +1,143 @@
+package com.box.l10n.mojito.monitoring;
+
+import com.box.l10n.mojito.Application;
+
+import org.awaitility.core.ConditionFactory;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.quartz.Job;
+import org.quartz.JobDetail;
+import org.quartz.JobExecutionContext;
+import org.quartz.Scheduler;
+import org.quartz.Trigger;
+import org.quartz.TriggerBuilder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.security.SecureRandom;
+import java.time.Duration;
+import java.time.ZonedDateTime;
+import java.util.Date;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.search.MeterNotFoundException;
+import io.micrometer.core.instrument.search.RequiredSearch;
+
+import static com.box.l10n.mojito.quartz.QuartzConfig.DYNAMIC_GROUP_NAME;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+import static org.quartz.JobBuilder.newJob;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = {Application.class},
```

Comment:
can we either piggy back on ServiceTestBase or have a simple test that doesn't bootstrap the whole application? 

I don't know if they changed the implementation since then, but at the time we use to have slow builds because of different test contexts (and/or using dirties context). So unless having a test context that is really limited in scope and quick to start, I'd rather avoid re-introducing that.

Reply:
Updated